source/SpinalHDL/Data types/bool.rst

```diff
@@ -9,7 +9,52 @@ Bool
 Description
 ^^^^^^^^^^^
 
-The ``Bool`` type corresponds to a boolean value (True or False).
```

Comment:
As mentioned in the chat - at least half of this should either go to the chapter on assignment or to to Introduction on Datatypes (and yes, we should fix the order that the Introduction comes first, but we can do that in a separate commit).
I'll still add some comments on the actual text.

Reply:
IMHO There are probably only around 5 major areas that are worth an explanation like this (a worked application of information to a specific domain) before the reader should have enough knowledge to understand reference style material.

Bool is defiantly one place.
Bits/Ints is another (which is conceptually a `bus`),
Mux/when/conditions is another,
Register is another
Combinational expressions, signals and signal direction another.

Maybe they would be improved in a tutorial/exercise form, with just a link/heading from the top of each relevant section with 'First time readers might with to work the tutorial on: Subject of tutorial' to gain a better understanding.

At the moment it is written in the form I wish I had seen to convey a number of concepts to think about at the same time, but my route was more examining the verilog output and working through the error messages :)